File: backend-systems-lab-stage-1/src/main/java/pl/jakubtworek/backend_systems_lab_stage_1/block_a/kotlin/coroutines/parallel/java/UserApiClient.java
package pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.coroutines.parallel.java;

import java.util.concurrent.ExecutorService;

// UserApiClient simulates loading user data from an external service.
public class UserApiClient {

    public User fetchUserById(String userId) {
        return new User(userId, "John Doe");
    }
}

File: backend-systems-lab-stage-1/src/main/java/pl/jakubtworek/backend_systems_lab_stage_1/block_a/kotlin/coroutines/parallel/kotlin/UserApiClient.kt
package pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.coroutines.parallel.kotlin

import kotlinx.coroutines.async
import kotlinx.coroutines.coroutineScope

// UserApiClient simulates loading user data from an external service.
class UserApiClient {

    suspend fun fetchUserById(userId: String): User {
        return User(userId, "John Doe")
    }
}